### PR TITLE
create generator for appropriate size string

### LINKED ISSUE #40: Update name slicer
There are names such as `MD Abdul Rahman Talukder`. These names will be converted into `MD` by the slicer. 
We need to update the slicer so that it can slice the name in a more readable manner.

### PR DESCRIPTION
**Pull request by csemanish12**
Closes #40 

created a generator for appropriate size string to handle names that exceeds max string length

### CODE REVIEW

**Comment by sudiptob2** [comment]
Seems good to me :+1: , can you also add unittests please, specially for the `__generate_appropriate_size_string`

**Comment by csemanish12** [comment]
added the units tests

File: tests/test_utils/test_string_utils.py [hunk 1]
```diff
@@ -49,3 +49,19 @@ def test_slice_with_long_name(self):
         actual = string_slicer.slice(full_name)
 
         assert expected == actual
+
+    def test_generate_appropriate_size_string(self):
+        full_name = "MD Abdul Rahman Talukder"
```
Comment (sudiptob2) [inline tests/test_utils/test_string_utils.py:54]: Please add docstring

File: tests/test_utils/test_string_utils.py [hunk 2]
```diff
@@ -49,3 +49,19 @@ def test_slice_with_long_name(self):
...
+        expected = "MD Abdul"
+        string_slicer = StringSlicer()
+        actual = string_slicer._StringSlicer__generate_appropriate_size_string(full_name)
+
+        assert expected == actual
+
+    def test_generate_appropriate_size_string_with_long_name(self):
+        full_name = "schwarzenegger"
```
Comment (sudiptob2) [inline tests/test_utils/test_string_utils.py:62]: docstring

**Review by sudiptob2 — COMMENTED** [review]
Please add some doctrings in the test functions

**Comment by sudiptob2** [comment]
Also, add your Name in the acknowledgement section of `Readme.md` ; Appreciate your effort :pray:

**Comment by csemanish12** [comment]
Updated the doc string in unit test
Updated the contributor list

**Comment by csemanish12** [comment]
Always happy to help

**Comment by sudiptob2** [comment]
Thanks; Merged

**Pull request merged**